### ISSUE TITLE
CUPS: PPD version is not compatible with Gutenprint

### ISSUE THREAD

**Issue by primeos** [NixOS/nixpkgs#28273]
## Issue description

After upgrading to ef026e02fcaa92323be875d8c7146b0b86e31880 I got the following error:
> "The PPD version (5.2.11) is not compatible with Gutenprint 5.2.13. Please run `/nix/store/7762kpyhfkcgmr3q81v1bbyy0bjhym80-gutenprint-5.2.13/sbin/cups-genppdupdate' as administrator." - http://localhost:631/jobs/

This was preventing me from printing anything with manual interaction. Executing the following command fixed it:
```bash
# /nix/store/7762kpyhfkcgmr3q81v1bbyy0bjhym80-gutenprint-5.2.13/sbin/cups-genppdupdate -p /etc/cups/ppd
Updated /etc/cups/ppd/XYZ.ppd using gutenprint.5.2://hp-lj_4350/expert
Updated 1 PPD file, 1 skipped. 
```

It seems like CUPS is keeping a copy of the old PPD file in `/etc/cups/ppd` and we don't automatically update these files (e.g. with the command above). Is this something we could fix or am I using CUPS in an unintended way (I'm not really familiar with CUPS...)?

I assume this error was caused by #27867, but that commit should be fine and don't cause any problems if we could somehow trigger an update of these PPD files.

### Steps to reproduce

Unconfirmed:
- Build 4f66c0cf21613301f0932b93ee885cab7f933fa7
- Add a printer via `http://localhost:631/admin` and use a PPD file from `Gutenprint v5.2.11` (in my case `HP LaserJet 4350 - CUPS+Gutenprint v5.2.13`)
- Optional: Try sending a job to the printer.
- Build 30f96ed2ba2bd78dd46d49177fad63097b41e60a
- Try sending a job to the printer.

My config:
```nix
  services = {
    printing = {
      enable = true;
      drivers = with pkgs; [ gutenprint hplip ];
    };
  };
```

cc @edolstra  @abbradar @cillianderoiste (CUPS & Gutenprint contributors)

**Comment by abbradar** [comment]
@primeos We are aware of this problem; because of this you need instead to set `printing.gutenprint = true` which adds autoupdating of PPD drivers.

Maybe we can somehow detect `gutenprint ` in drivers instead? For example, `meta.gutenprint = true`.

**Comment by primeos** [comment]
@abbradar Thanks, I wasn't aware of this and it seems like I missed the note in the description... (or I haven't looked at it since d37729f4b8978af3ed2776cf706884f116997fb5)

Imho automatically detecting a `gutenprint` derivation in the drivers list is a good idea (assuming there isn't any real reason why one would not want auto-updating of the PPD files).

**Comment by primeos** [comment]
Would something like #28423 be ok or is that a bad idea/approach?

**Comment by primeos** [comment]
Should be fixed by ea23f8bb079ef11c2d11859994d804c4e4940be5.
